src/fmain.pas

```diff
@@ -6711,6 +6727,80 @@ procedure Tfrmain.milistPropertiesClick(Sender: TObject);
       ///////CONFIRM DIALOG MODE///////////
       if frmain.lvMain.SelCount>1 then
       begin
+        // collecting the selected items' information
```

Comment:
In general, I'd rather extract this as a helper method - or maybe introduce two helper methods: the first one would prepare the frnewdown's dialog, basing on the information collected from the selected items; and the second one would apply the frnewdown's dialog's properties either to selected items or to a single item. The purpose would be just to have several smaller methods rather than one big method.

Reply:
you are right